powerstore/resource_storagecontainer.go

```diff
@@ -154,7 +163,73 @@ func (r resourceStorageContainer) Read(ctx context.Context, req tfsdk.ReadResour
 
 // Update resource
 func (r resourceStorageContainer) Update(ctx context.Context, req tfsdk.UpdateResourceRequest, resp *tfsdk.UpdateResourceResponse) {
+	log.Printf("Started Update")
+
+	// Get plan values
+	var plan models.StorageContainer
+	diags := req.Plan.Get(ctx, &plan)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// Get current state
+	var state models.StorageContainer
+	diags = req.State.Get(ctx, &state)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// a worksround
+	// currently a bug on powerstore, not accepting PATCH call for same values
+
+	storageContainerUpdate := &gopowerstore.StorageContainer{}
+
+	if plan.Name.Value != state.Name.Value {
+		storageContainerUpdate.Name = plan.Name.Value
+	}
+
+	if plan.Quota.Value != state.Quota.Value {
+		storageContainerUpdate.Quota = plan.Quota.Value
+	}
+
+	if plan.StorageProtocol.Value != state.StorageProtocol.Value {
+		storageContainerUpdate.StorageProtocol = gopowerstore.StorageContainerStorageProtocolEnum(plan.StorageProtocol.Value)
+	}
```

Comment:
How about update of high_water_mark value, did you check its feasibility? since we will have to document the same if not feasible